ui/component/common/file-selector.jsx

```diff
@@ -64,13 +66,26 @@ class FileSelector extends React.PureComponent<Props> {
       properties = ['openDirectory'];
     }
 
-    remote.dialog.showOpenDialog({ properties, defaultPath }).then((result) => {
-      const path = result && result.filePaths[0];
-      if (path) {
-        // $FlowFixMe
-        this.props.onFileChosen({ path });
-      }
-    });
+    remote.dialog
+      .showOpenDialog({
+        properties,
+        defaultPath,
+        filters: this.props.filters,
+      })
+      .then((result) => {
+        const path = result && result.filePaths[0];
+        if (path) {
+          return ipcRenderer.invoke('get-file-from-path', path);
+        }
+        return undefined;
+      })
+      .then((result) => {
+        if (!result) {
+          return;
+        }
+        const file = new File([result.buffer], result.name);
+        this.props.onFileChosen(file);
```

Comment:
I'm concerned that changing this from { path } to file will break publishing. Did you try?
https://github.com/lbryio/lbry-desktop/blob/b9be8d9f3a80d06c3e620ad12fc692f907c46223/ui/component/publishFile/view.jsx#L273 expects file.path on electron because the sdk needs the path.

It may be good to send { file: file , path: path } and adjust all the functions assigned to onFileChosen to take that.

Reply:
Sounds good @jessopb, nice catch. Perhaps as a small note for the future, we should avoid using these union types. Notice how I'm still respecting the interface (by sending a `File` type) but still, it manages to break parts of the app...

Reply:
You're probably right. The union types were originally added when the codebase was supporting both web and app. 

Maybe the type should be ChosenFile = { file: WebFile, path: string }

Reply:
@jessopb I have updated the code so the `File` now contains a `path` and a `mime` type. With this, the file upload for posts works fine. Please let me know what you think. Thanks.